server/api/src/main/java/org/hawkular/apm/server/api/services/PropertyService.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2015-2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.hawkular.apm.server.api.services;
+
+/**
+ * Provides cache for a property domain class. This class also supports access to the system and
+ * environmental properties through {@link Cache#get(String, String)} method. If it finds the property it
+ * stores it in the cache.
+ *
+ * @author Pavol Loffay
+ */
+public interface PropertyService extends Cache<Property<?>> {
+
+    /**
+     * Store single property in the cache.
+     *
+     * @param tenantId tenant
+     * @param property property
+     * @throws CacheException
+     */
+    void store(String tenantId, Property<?> property) throws CacheException;
+
+    /**
+     * Add observer for given id. Method {@link Observer#update(Property, Property)} if called when client
+     * invokes This{@link #store(String, Property)} on property which have been already stored in the cache.
```

Comment:
Comment is a bit misleading. First I think you mean "is called when ...". However the update callback isn't called when store called on property already stored in cache - it would be possible for an observer to be registered before a property exists in the cache, and the update would be called when that property is initially written using the store() method.